pkg/lightning/errormanager/errormanager.go

```diff
@@ -149,22 +151,23 @@ const (
 	sqlValuesConflictErrorIndex = "(?,?,?,?,?,?,?,?,?,?)"
 
 	selectIndexConflictKeysReplace = `
-		SELECT raw_key, index_name, raw_value, raw_handle
+		SELECT _tidb_rowid, raw_key, index_name, raw_value, raw_handle
 		FROM %s.` + ConflictErrorTableName + `
-		WHERE table_name = ? AND is_data_kv = 0
-		ORDER BY raw_key;
+		WHERE table_name = ? AND is_data_kv = 0 AND _tidb_rowid >= ? and _tidb_rowid < ?
+		ORDER BY _tidb_rowid LIMIT ?;
 	`
 
 	selectDataConflictKeysReplace = `
-		SELECT raw_key, raw_value
+		SELECT _tidb_rowid, raw_key, raw_value
 		FROM %s.` + ConflictErrorTableName + `
-		WHERE table_name = ? AND is_data_kv = 1
-		ORDER BY raw_key;
+		WHERE table_name = ? AND is_data_kv = 1 AND _tidb_rowid >= ? and _tidb_rowid < ?
+		ORDER BY _tidb_rowid LIMIT ?;
 	`
 
 	deleteNullDataRow = `
 		DELETE FROM %s.` + ConflictErrorTableName + `
-		WHERE key_data = "" and row_data = "";
+		WHERE key_data = ""
```

Comment:
maybe change `is_data_kv` to a `type varchar` field with it's values `data` `index` `temp`, and add index for it, so we can query/delete them faster

Reply:
As discussed, we will use `kv_type` to determine three types of kv:
0 for index kv, 1 for data kv, 2 for additionally inserted data kv

The index for kv_type` has been added to ensure faster deletion.